kong/plugins/bot-detection/handler.lua

```diff
@@ -11,6 +12,19 @@ local BotDetectionHandler = BasePlugin:extend()
 
 BotDetectionHandler.PRIORITY = 2500
 
+local MATCH_EMPTY     = 0
+local MATCH_WHITELIST = 1
+local MATCH_BLACKLIST = 2
+local MATCH_BOT       = 3
+
+-- per-worker cache of matched UAs
+-- we could use the kong cache mechanism, but purging a subset of
+-- the cache is a pain, and the lookup here isn't so expensive that
+-- we really worry about it. so we just cache per worker and invalidate
+-- when the plugin has been updated
+local ua_cache = {}
+local UA_CACHE_SIZE = 10 ^ 4
```

Comment:
Definitely fine with either notation, but what do we think of the following notation when defining such constants going forward: `local UA_CACHE_SIZE = 1e4`?